pkg/types/config/interfaces.go

```diff
@@ -0,0 +1,81 @@
+package config
+
+import (
+	corecontrollers "github.com/rancher/wrangler/v3/pkg/generated/controllers/core"
+	core "github.com/rancher/wrangler/v3/pkg/generated/controllers/core/v1"
+	wrbac "github.com/rancher/wrangler/v3/pkg/generated/controllers/rbac/v1"
+	"github.com/rancher/wrangler/v3/pkg/generic"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/cache"
+)
+
+// nonNamespacedCacheAndClient limits a Controller interface to only the direct and cached clients
+type nonNamespacedCacheAndClient[T generic.RuntimeMetaObject, TList runtime.Object] interface {
+	generic.NonNamespacedClientInterface[T, TList]
+	Cache() generic.NonNamespacedCacheInterface[T]
+	Informer() cache.SharedIndexInformer // needed to immediately start the caches
+}
+
+// namespacedCacheAndClient limits a Controller interface to only the direct and cached clients
+type namespacedCacheAndClient[T generic.RuntimeMetaObject, TList runtime.Object] interface {
+	generic.ClientInterface[T, TList]
+	Cache() generic.CacheInterface[T]
+	Informer() cache.SharedIndexInformer // needed to immediately start the caches
+}
+
+// rbacInterface does not restrict any usage of the original interface, as Steve's accesscontrol uses caches and indexers for all resources
+type rbacInterface wrbac.Interface
+
+// coreInterface provides a reduced interface from the regular core.Interface
+// This prevents new caches being added unintentionally.
+type coreInterface struct {
+	factory *corecontrollers.Factory
+}
+
+// Secret provides access only for a downstream client.
+//
+// WARNING: no caches or shared controller should be provided here, as that implies caching all secrets from every downstream cluster.
+//
+//	Please check with the Bullseye team before modifying this interface.
+func (i coreInterface) Secret() core.SecretClient {
+	return i.factory.Core().V1().Secret()
+}
```

Comment:
Why does this function return a "Client" while others return a "Controller"?  
Is it possible to update all places where this function is called to accept a "Controller" so that we can unify the pattern here?